src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -581,4 +583,36 @@ private static void assertNotNull(AssertionInfo info, Class<?> actual) {
   private static void classParameterIsNotNull(Class<?> clazz) {
     requireNonNull(clazz, "The class to compare actual with should not be null");
   }
+
+  /**
+   * Verifies that the actual {@code Class} has the given {@code packagename}.
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param packageName the package that must be declared in the class.
+   */
+  public void assertHasPackage(AssertionInfo info, Class<?> actual, String packageName) {
+    assertNotNull(info, actual);
+    requireNonNull(packageName, shouldNotBeNull("package").create());
+    Package actualPackage = actual.getPackage();
+
+    if (actualPackage == null || !actualPackage.getName().equals(packageName)) {
```

Comment:
The `actualPackage == null` check made me think that for completeness we should have another `hasNoPackage()` assertion, which would cover the cases mentioned in the [`getPackage()`](https://docs.oracle.com/javase/9/docs/api/java/lang/Class.html#getPackage--) javadoc: array types, primitive types and void.

However, I wouldn't include `hasNoPackage()` in this PR but as a separate change. I will create an issue to track it.

Reply:
Oh right, didn't think about that! I tried to follow the Javadoc closely but didn't try to cover those cases. Thanks for raising the issue, maybe I'll work on it as soon as I can.

Reply:
> However, I wouldn't include `hasNoPackage()` in this PR but as a separate change. I will create an issue to track it.

Hey @scordio I don't know if you created this but I would like to contribute this assertion too if you don't mind, thanks.